packages/@aws-cdk/aws-glue/lib/database.ts

```diff
@@ -83,27 +82,29 @@ export class Database extends Resource implements IDatabase {
   /**
    * Location URI of this database.
    */
-  public readonly locationUri: string;
+  public readonly locationUri?: string;
 
   constructor(scope: Construct, id: string, props: DatabaseProps) {
     super(scope, id, {
       physicalName: props.databaseName,
     });
 
+    let databaseInput: CfnDatabase.DatabaseInputProperty = {
+      name: props.databaseName,
+    };
+
     if (props.locationUri) {
       this.locationUri = props.locationUri;
-    } else {
-      const bucket = new s3.Bucket(this, 'Bucket');
-      this.locationUri = `s3://${bucket.bucketName}/${props.databaseName}`;
+      databaseInput = {
+        locationUri: this.locationUri,
+        ...databaseInput
+      };
     }
 
     this.catalogId = Stack.of(this).account;
     const resource = new CfnDatabase(this, 'Resource', {
       catalogId: this.catalogId,
-      databaseInput: {
-        name: this.physicalName,
-        locationUri: this.locationUri
-      }
+      databaseInput: databaseInput
```

Comment:
I believe you need to change this to just `databaseInput` to silence the linter that's making the build fail.